pkg/urfave/urfave.go

```diff
@@ -0,0 +1,200 @@
+// Copyright (c) 2019, the Drone Plugins project authors.
+// Please see the AUTHORS file for details. All rights reserved.
+// Use of this source code is governed by an Apache 2.0 license that can be
+// found in the LICENSE file.
+
+// Package urfave provides helpers for interacting with the `urfave/cli`
+// package when creating plugins for use by the Drone CI/CD service.
+//
+// Drone communicates to plugins by passing in environment variables that have
+// information on the currently executing build. The `urfave/cli` package can
+// read these environment variables and extract them into structs.
+//
+//     import(
+//         "github.com/drone-plugins/drone-plugin-lib/pkg/urfave"
+//         "github.com/urfave/cli"
+//	   )
+//
+//     func main() {
+//         app := cli.New()
+//         app.Name = "my awesome Drone plugin"
+//         app.Run = run
+//         app.Flags = []cli.Flags{
+//             // All my plugin flags
+//         }
+//         app.Flags = append(
+//             app.Flags,
+//             urfave.CommitFlags()...,
+//         )
+//     }
+package urfave
+
+import (
+	"time"
+
+	"github.com/drone-plugins/drone-plugin-lib/pkg/plugin"
+	"github.com/urfave/cli"
+)
+
+//---------------------------------------------------------------------
+// Stage Flags
+//---------------------------------------------------------------------
+
+const (
+	// StageArchFlag corresponds to plugin.Stage.Arch.
+	StageArchFlag = "DRONE_STAGE_ARCH"
+	// StageDependsOnFlag corresponds to plugin.Stage.DependsOn.
+	StageDependsOnFlag = "DRONE_STAGE_DEPENDS_ON"
+	// StageFinishedFlag corresponds to plugin.Stage.Finished.
+	StageFinishedFlag = "DRONE_STAGE_FINISHED"
+	// StageKindFlag corresponds Stage.Kind.
+	StageKindFlag = "DRONE_STAGE_KIND"
+	// StageMachineFlag corresponds to plugin.Stage.Machine.
+	StageMachineFlag = "DRONE_STAGE_MACHINE"
+	// StageNameFlag corresponds to plugin.Stage.Name.
+	StageNameFlag = "DRONE_STAGE_NAME"
+	// StageNumberFlag corresponds to plugin.Stage.Number.
+	StageNumberFlag = "DRONE_STAGE_NUMBER"
+	// StageOSFlag corresponds to plugin.Stage.OS.
+	StageOSFlag = "DRONE_STAGE_OS"
+	// StageStartedFlag corresponds to plugin.Stage.Started.
+	StageStartedFlag = "DRONE_STAGE_STARTED"
+	// StageStatusFlag corresponds to plugin.Stage.Status.
+	StageStatusFlag = "DRONE_STAGE_STATUS"
+	// StageTypeFlag corresponds to plugin.Stage.Type.
+	StageTypeFlag = "DRONE_STAGE_TYPE"
+	// StageVariantFlag corresponds to plugin.Stage.Variant.
+	StageVariantFlag = "DRONE_STAGE_VARIANT"
+	// StageVersionFlag corresponds to plugin.Stage.Version.
+	StageVersionFlag = "DRONE_STAGE_VERSION"
```

Comment:
This environment variable is aspirational since it isnt there but it would be useful for winders builds.